src/PowerShellEditorServices/Services/DebugAdapter/DebugStateService.cs

```diff
@@ -25,8 +31,19 @@ internal class DebugStateService
 
         internal bool IsInteractiveDebugSession { get; set; }
 
-        internal bool SetBreakpointInProgress { get; set; }
+        // If the CurrentCount is equal to zero, then we have some thread using the handle.
+        internal bool IsSetBreakpointInProgress => _setBreakpointInProgressHandle.CurrentCount == 0;
 
         internal bool IsUsingTempIntegratedConsole { get; set; }
+
+        internal void ReleaseSetBreakpointHandle()
+        {
+            _setBreakpointInProgressHandle.Release();
+        }
+
+        internal async Task WaitForSetBreakpointHandleAsync()
+        {
+            await _setBreakpointInProgressHandle.WaitAsync();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4546488)